apps/rush-lib/src/logic/PackageChangeAnalyzer.ts

```diff
@@ -56,10 +58,32 @@ export class PackageChangeAnalyzer {
     let repoDeps: IPackageDeps;
     try {
       if (this._isGitSupported) {
+        const commandLineConfigFile: string = path.join(
+          this._rushConfiguration.commonRushConfigFolder,
+          RushConstants.commandLineFilename
+        );
+
+        const commandLineConfiguration: CommandLineConfiguration = CommandLineConfiguration.loadFromFileOrDefault(
+          commandLineConfigFile
+        );
+
+        // Generate possible package-deps filenames from commands in order to
+        // ensure they are not included in the repoDeps
+        const excludes: string[] = commandLineConfiguration.commands.reduce(
+          (filenames: string[], command: CommandJson) => {
+            if (command.commandKind === 'bulk') {
+              filenames.push(`package-deps.${command.name}.json`);
```

Comment:
Doesn't your change include putting all of these under a `.rush` folder? Wouldn't it be easier to just exclude that folder?

Reply:
hmm, it probably makes sense to ignore the `.rush` folder completely otherwise running a different incremental command will create the file `package-deps.different.json`, which will then invalidate `package-deps.build.json` causing projects to be rebuilt unnecessarily.

Reply:
The different package-deps don't cause each other to invalidate by design in order to ensure, for example, that incremental changes to code don't cause build missed just because tests were run. In order to do that, we're simply pushing all of the known commands to the excludes list. If there's a quick and easy way to instead ignore an entire folder, then that would be easier, but reading through the package-deps-hash code, it didn't seem like providing a folder name would produce the expected outcome.

Reply:
We're talking about .gitignore are we not?

I think ignoring each package-deps*.json file will suffice. directories don't mean anything in git world AFAIK.

Ignoring the .rush directory could work too, but as a user, I'd react something like..

>User: What are these new files ready to be committed?? Oh, those are just package-deps.json files, I'll ignore those by A) .gitignoring the .rush directory, which I as a user am not sure about if that's correct, or B) .gitignoring the new package-deps files, which feels safer. I will go with plan B, unless the docs say anything different.

If a .rush directory fits other designs later on, we could update the docs and best practises to ignore them.

Personally I'm not a great fan of the name `.rush`, would prefer `.rush-temp`, or a `.rush/temp` structure that is more aligned with the `common/rush/temp` structure in the root.